src/c4/test/tstifpstream.cc

```diff
@@ -0,0 +1,161 @@
+//----------------------------------*-C++-*-----------------------------------//
+/*!
+ * \file   c4/test/tstifpstream.cc
+ * \author Mathew Cleveland
+ * \date   Wed Apr 28 09:31:51 2010
+ * \brief  Tests ifpstream class for paralle read
+ * \note   Copyright (C) 2016-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//----------------------------------------------------------------------------//
+
+#include "c4/ParallelUnitTest.hh"
+#include "c4/ifpstream.hh"
+#include "c4/ofpstream.hh"
+#include "ds++/Release.hh"
+#include <sstream>
+
+using namespace std;
+using namespace rtt_dsxx;
+using namespace rtt_c4;
+
+//----------------------------------------------------------------------------//
+// TESTS
+//----------------------------------------------------------------------------//
+
+void write_stream_ascii() {
+
+  unsigned const pid = rtt_c4::node();
+  ofpstream out("tstifpstream.txt");
+
+  if (pid == 0)
+    out << "just rank 0" << endl;
+
+  out << "MPI rank " << pid << " reporting ..." << endl;
+  out.send();
+  out.shrink_to_fit();
+
+  out << "MPI rank " << pid << " reporting a second time ..." << endl;
+  out.shrink_to_fit();
+  out.send();
+}
+
+void write_stream_binary() {
```

Comment:
Dumb question: What actually happens when you put ascii data in the binary stream here? I assume you'd have to do a reinterpret_cast<char *>, but maybe I'm wrong. If I AM wrong, maybe it would be worthwhile to collapse the helper functions into something like write_stream(mode), since most of the body is duplicated. :)

Reply:
binary simple specifies the formate of the file read/write (not the data coming in/out). As you pointed I could combine these into a single function, which I have now done.